packages/ag-charts-website/src/features/codeSandbox/utils/codeSandbox.ts

```diff
@@ -0,0 +1,159 @@
+    fileName,
+    internalFramework,
+}: {
+    fileName: string;
+    internalFramework: InternalFramework;
+}) => {
+    if (!isReactFramework(internalFramework)) {
+        return fileName;
+    }
+
+    if (fileName === 'index.html') {
+        return `public/index.html`;
+    }
+
+    if (/([a-zA-Z0-9\\s_.])+(.js|.jsx|.tsx|.ts|.css)$/.test(fileName)) {
+        if (fileName.endsWith('.js')) {
+            return `public/${fileName}`;
+        }
+
+        if (fileName.startsWith('index.')) {
+            return `src/${fileName === 'index.jsx' ? 'index.js' : fileName}`;
+        }
+
+        if (fileName === 'styles.css') {
+            return `src/styles.css`;
+        }
+        return `src/${fileName}`;
+    }
+
+    return fileName;
+};
+
+const getCodeSandboxRuntime = (internalFramework: InternalFramework) => {
+    switch (internalFramework) {
+        case 'reactFunctional':
+            return 'create-react-app';
+        case 'reactFunctionalTs':
+            return 'create-react-app-typescript';
+        default:
+            return 'static';
+    }
+};
+
+const getCodeSandboxFiles = ({
+    files,
+    boilerPlateFiles,
+    internalFramework,
+}: {
+    files: FileContents;
+    boilerPlateFiles: FileContents;
+    internalFramework: InternalFramework;
+}) => {
+    const sandboxFiles: SandboxFiles = {};
+    const allFiles = isReactFramework(internalFramework)
+        ? {
+              ...files,
+          }
+        : { ...boilerPlateFiles, ...files };
+
+    for (const [name, content] of Object.entries(allFiles)) {
+        const key = getPathForFile({ fileName: name, internalFramework });
+        sandboxFiles[key] = {
+            content: content as string,
+        };
+    }
+
+    return sandboxFiles;
+};
+
+const getCodeSandboxBody = ({
+    title,
+    files,
+    boilerPlateFiles,
+    internalFramework,
+}: {
+    title: string;
+    files: FileContents;
+    boilerPlateFiles: FileContents;
+    internalFramework: InternalFramework;
+}) => {
+    const runtime = getCodeSandboxRuntime(internalFramework);
+    const configFiles = {
+        '.codesandbox/template.json': {
+            content: JSON.stringify(getCodeSandboxTemplate({ title, runtime }), null, 2),
+        },
+    };
+    const sandboxFiles = {
+        ...configFiles,
+        ...getCodeSandboxFiles({
+            files,
+            boilerPlateFiles,
+            internalFramework,
+        }),
+    };
+
+    return {
+        files: sandboxFiles,
+    };
+};
+
+export const createNewCodeSandbox = async ({
+    title,
+    files,
+    boilerPlateFiles,
+    internalFramework,
+}: {
+    title: string;
+    files: FileContents;
+    boilerPlateFiles: FileContents;
+    internalFramework: InternalFramework;
+}) => {
+    const bodyObj = getCodeSandboxBody({
+        title,
+        files,
+        boilerPlateFiles,
+        internalFramework,
+    });
+    const body = JSON.stringify(bodyObj);
+
+    return fetch(CREATE_CODE_SANDBOX_URL, {
```

Comment:
Could use await here - but not too important

Reply:
I was thinking, if it's not awaited, it can be left up to the caller to decide how they want to do the await